cactus/page.py

```diff
@@ -65,26 +66,27 @@ def full_build_path(self):
         return os.path.join(self.site.build_path, self.build_path)
 
     def data(self):
+        if not hasattr(self, "_data"):
```

Comment:
I'd rather you set `_data` to `None` (either in `__init__` or in the `class` definition) and check for `None`. Functionally equivalent an plays much better with with refactoring tools.